core/src/main/java/org/apache/accumulo/core/fate/zookeeper/MetaFateStore.java

```diff
@@ -202,25 +213,27 @@ public void deleteDeadReservations() {
       if (isLockHeld.test(reservation.getLockID())) {
         continue;
       }
+
+      UnaryOperator<FateData<T>> fateDataOp = currFateData -> {
+        // Make sure the current node is still reserved and reserved with the expected reservation
+        // and it is dead
+        if (currFateData.isReservedBy(reservation)
+            && !isLockHeld.test(currFateData.reservation.orElseThrow().getLockID())) {
+          // Delete the reservation
+          log.trace("Deleted the dead reservation {} for fate id {}", reservation, fateId);
+          return new FateData<>(currFateData.status, null, currFateData.fateKey.orElse(null),
+              currFateData.repoDeque, currFateData.txInfo);
+        } else {
+          // This will not change the value and will return null
+          return null;
+        }
+      };
+
       try {
-        zk.mutateExisting(getTXPath(fateId), currSerNodeVal -> {
-          NodeValue currNodeVal = new NodeValue(currSerNodeVal);
-          // Make sure the current node is still reserved and reserved with the expected reservation
-          // and it is dead
-          if (currNodeVal.isReservedBy(reservation)
-              && !isLockHeld.test(currNodeVal.reservation.orElseThrow().getLockID())) {
-            // Delete the reservation
-            log.trace("Deleted the dead reservation {} for fate id {}", reservation, fateId);
-            return new NodeValue(currNodeVal.status, null, currNodeVal.fateKey.orElse(null))
-                .serialize();
-          } else {
-            // No change
-            return null;
-          }
-        });
-      } catch (KeeperException.NoNodeException e) {
+        mutate(fateId, fateDataOp);
+      } catch (KeeperException.NoNodeException nne) {
         // the node has since been deleted. Can safely ignore
-      } catch (KeeperException | InterruptedException | AcceptableThriftTableOperationException e) {
+      } catch (KeeperException e) {
         throw new RuntimeException(e);
       }
```

Comment:
With new structure with new `mutate()` method, this will only throw `RuntimeException` if it is a `KeeperException` and `IllegalStateException` otherwise... Previously was throwing RuntimeException for all cases... not sure if it matters to keep the RuntimeException for InterruptedException or AcceptableThriftTableOperationException, but thought I'd mention it in case it does

Reply:
This seems ok